fetch_index.py

```diff
@@ -0,0 +1,354 @@
+                if not response:
+                    logger.error(
+                        'Could not fetch the items for list: {}'.format(list_content))
+                    return []
+                try:
+                    response_data = response.json()
+                    response_data = response_data.get('d', {}).get('results')
+                except Exception as exception:
+                    logger.error('Error while parsing the get items response for list: {} from url: {}. Error: {}'.format(
+                        list_content, rel_url, exception))
+                    return []
+                logger.info(
+                    'Successfuly fetched and parsed the listitem response for list: {} from SharePoint'.format(list_content))
+
+            else:
+                logger.info(
+                    'The list: {} contains some files/folder'.format(list_content))
+                # TODO: Change the rel_url and query to fetch the file/folder contents
+                response_data = self.sharepoint_client.get(rel_url, self.query)
+                # TODO: need to add logic to index text
+                text = extract(response_data)
+
+            for index in range(len(response_data)):
+                response_data[index] = dict((k.lower(), v)
+                                            for k, v in response_data[index].items())
+                response_data[index].pop('__metadata')
+
+            if index:
+                json_object = json.dumps(response_data, indent=4)
+                logger.info(
+                    'Indexing the listitem for list: {} to the Workplace'.format(list_content))
+                try:
+                    self.ws_client.index_documents(
+                        http_auth=self.ws_token,
+                        content_source_id=self.ws_source,
+                        documents=json_object
+                    )
+                    logger.info(
+                        'Successfully indexed the listitem for list: {} to the workplace'.format(list_content))
+                except Exception as exception:
+                    logger.error('Error while indexing the listitem for list: {} to the workplace. Error: {}'.format(
+                        list_content, exception))
+                    self.is_error = True
+                    return []
+            responses.append(response_data)
+        return responses
+
+    def get_items_ids(self, lists, response_data=None):
+        logger.info('Extracting items ids')
+        items = {}
+        if not response_data:
+            logger.info(
+                'ListItem response is not present. Fetching the items for the list')
+            response_data = self.index_items(lists, index=False) or []
+
+        items = {}
+        listitems = []
+        for response in response_data:
+            for result in response:
+                listitems.append(result.get('id'))
+            items[result.get('title')] = listitems
```

Comment:
Not sure I understand this part. `listitems` is defined outside of these two `for` loops and so it's always the same.
It looks like all the values of the `items` dictionary always reference the same list?
Also what is the value of `result` here? The last value it was assigned from the previous `for` loop?

Reply:
Hey @ioanatia, thanks for flagging that. The issue has been resolved.